### ISSUE TITLE
Feature request: Support for preview comment.

### ISSUE THREAD

**Issue by NorbertBodziony** [cloudflare/wrangler-action#77]
Would love to get support for preview comment after successful deployment.
Something like this https://github.com/BetaHuhn/deploy-to-vercel-action for vercel.

**Comment by NullVoxPopuli** [comment]
Or maybe like Cloudflare Pages?

example: https://github.com/NullVoxPopuli/ember-resources/pull/529

![image](https://user-images.githubusercontent.com/199018/173237400-e1e59855-adde-41bb-8873-a801ed11f003.png)

**Comment by GrantBirki** [comment]
Yes please! This feature would be awesome. Or even a way for us to construct that message via exported values. This Action currently takes inputs but does not save any outputs.

I tried to manually construct this message by saving the `.stdout` of this Action but no luck. This feature would be greatly appreciated

**Comment by adam-clio** [comment]
Came here to request this feature. Even some output from this action would be awesome so we could create our own comments

**Comment by danawoodman** [comment]
Hate to be "that guy", but can we get a bumpbump on this bad boy? Kinda makes this action worthless for PR deploys at this point.

Also, seems related to #90 #81

**Comment by NorbertBodziony** [comment]
Hey @JacobMGEvans can you provide more info why this issue has been closed ?

**Comment by JacobMGEvans** [comment]
Had closed them all, this sounds like a great idea. Will revisit it, thanks for the follow up!

**Comment by JacobMGEvans** [comment]
If it Pages related I suggest this comment as well to help https://github.com/cloudflare/wrangler-action/issues/65#issuecomment-1670136345

**Comment by hellt** [comment]
It would be great if wrangler action would provide the same bot functionality as shown by the OP.

**Comment by srl295** [comment]
I did this myself like so:

https://github.com/unicode-org/cldr/blob/9597d2390dfe66b8a892d7ec60543c3e6007c6f3/.github/workflows/site.yml#L49-L58

**Comment by hellt** [comment]
This is sweet, thanks @srl295 
The missing piece, though, is having an alias with the branch name, which seems to be totally gone when the static artifacts are pushed via wrangler and not via the direct repo integration

**Comment by srl295** [comment]
> This is sweet, thanks @srl295 The missing piece, though, is having an alias with the branch name, which seems to be totally gone when the static artifacts are pushed via wrangler and not via the direct repo integration

I'm not 24 hours into my integration at this point, so..  I don't quite follow what you mean.  You want  "mysite-mybranch.pages.dev" or something?  I don't quite see how that's part of this issue - that sounds like an option on the upload side?

**Comment by hellt** [comment]
> You want "mysite-mybranch.pages.dev" or something? I don't quite see how that's part of this issue - that sounds like an option on the upload side?

Yes, this is about Branch Prefiew URL as shown in https://github.com/cloudflare/wrangler-action/issues/77#issuecomment-1153183792

I agree, it might be considered not related to the request of having ability for an action to post a comment, if we only consider comment to have a hash-based preview URL that is visible in the action output.

**Comment by srl295** [comment]
the current action produces a single URL as output. So it seems like something else is needed if you want a different URL.  Maybe you should file a separate issue for it?

**Comment by ve1997** [comment]
I'm also looking for this feature.
I had no choice but to write my own workflow, so there's no problem, but if it's officially supported, that would be best.

**Comment by kleinpetr** [comment]
still with no update here? for almost 3 years?..

**Comment by moojing** [comment]
I just added an action to easily view the preview url using the same format as @namidapoo.
https://github.com/marketplace/actions/comment-cloudflare-preview-url-on-pr
Someone want to help me to test if it's easy to use for anyone?

**Comment by andypandy-tappit** [comment]
We used the following steps in our workflows to achieve this for anyone interested. It will generate a similar PR comment to the cloudflare plugin with the relevant preview and alias URL's. The comment is tagged so any updates are applied in place preventing new comments being added onto your PR with each push. Plus it generates a step output the same as the cloudflare plugin. I don't reference step output very much but it does align with functionality provided by the plugin!
```
      - name: Deploy Preview
        id: deploy_preview
        uses: cloudflare/wrangler-action@v3
        env:
          COMMIT_MESSAGE: ${{ steps.get_head_commit_message.outputs.COMMIT_MESSAGE }}
        with:
          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
          command: pages deploy dist --project-name={your project name here} --branch=${{ env.BRANCH_NAME }} --commit-hash=${{ github.sha }} --commit-message="${{ env.COMMIT_MESSAGE }}"

      - name: Comment deployment-url
        uses: thollander/actions-comment-pull-request@v3
        env:
          DEPLOYMENT_URL: ${{ steps.deploy_preview.outputs.deployment-url }}
          DEPLOYMENT_ALIAS_URL: ${{ steps.deploy_preview.outputs.pages-deployment-alias-url }}
        with:
          comment-tag: preview
          message: |
            ## Deploying {your project name here} with Cloudflare Pages
            | Status  | ✅ Deploy Successful! |
            | :------------ | :------------ |
            | Preview Url:  |  ${{ steps.deploy_preview.outputs.deployment-url }} |
            | Branch Preview Url:  | ${{ steps.deploy_preview.outputs.pages-deployment-alias-url }}  |

      - name: Generate Summary
        run: |
          echo "## Deploying {your project name here} with Cloudflare Pages" >> $GITHUB_STEP_SUMMARY
          echo "" >> $GITHUB_STEP_SUMMARY
          echo "| Status  | ✅ Deploy Successful! |" >> $GITHUB_STEP_SUMMARY
          echo "| :------------ | :------------ |" >> $GITHUB_STEP_SUMMARY
          echo "| Preview Url:  | ${{ steps.deploy_preview.outputs.deployment-url }} |" >> $GITHUB_STEP_SUMMARY
          echo "| Branch Preview Url:  | ${{ steps.deploy_preview.outputs.pages-deployment-alias-url }} |" >> $GITHUB_STEP_SUMMARY
```

**Comment by danawoodman** [comment]
do any of these solutions post a comment with all deployed workers apps for a monorepo? 

we have a monorepo that has multiple apps each with different envs that are each setup to deploy based on the targeted branch and a file pattern. 

if, for example, i was to update our bun.lockb file on a PR, it should deploy like 6 different workers for our various apps for our "dev" envs we have setup. 

how could we make sure the PR comment has the deploy URL of every app that was triggered to be deployed for the given PR?

seems like the official CF action does not handle this and just randomly shows one of the 6 apps in the comment with no rhyme or reason as to which.

**Comment by marcustyphoon** [comment]
I haven't fully tested this, but I just pushed a deployment in a PR using `cloudflare/wrangler-action@v3` and _did_ [get a preview comment of this type](https://github.com/discretize/discretize-gear-optimizer/pull/1003), I believe due to the organization having the Cloudflare github app installed. Interestingly, this seems to happen if and only if one deploys the PR head commit rather than the PR merge commit.

<details><summary>i.e.</summary>

```yaml
name: Deploy

on:
  pull_request:
    types: [opened, reopened, synchronize, ready_for_review]

jobs:
  deploy:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      deployments: write

    steps:
      - name: Checkout
        uses: actions/checkout@v6
        # this checks out the head, not the merge, so the result may be out of date
        with:
          ref: ${{ github.event.pull_request.head.sha }}

      - name: Build
        run: # whatever

      - name: Deploy
        uses: cloudflare/wrangler-action@v3
        with:
          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
          command: pages deploy YOUR_DIST_FOLDER --project-name=example --branch=${{ github.head_ref || github.ref_name }}
          gitHubToken: ${{ secrets.GITHUB_TOKEN }}
```
</details>

I assume this is kind of a bug/oversight/limitation in the Cloudflare github app, i.e. that the app is detecting which PRs to add comments to via head SHA?

Normally one would want to test and deploy the PR merge commit so that it reflects the latest changes to the target branch. (That's the default GITHUB_SHA in `pull_request` workflows and thus the default commit that gets checked out by `actions/checkout` with no arguments.)
